bbb-graphql-server/bbb_schema.sql

```diff
@@ -1970,13 +1974,8 @@ select "meeting"."meetingId",
         ) as "hasExternalVideo",
         exists (
             select 1
-            from "v_user"
-            where "v_user"."meetingId" = "meeting"."meetingId"
-            and NULLIF("speechLocale",'') is not null
```

Comment:
By observing `user.speechLocale` it will be easier to predict that some user might send a transcription (caption).

Reply:
> Which would make it impossible for the users to receive the first transcription.

Agreed, I've been thinking about it too, but it doesn't seem right, to me, to entangle the speechLocale with the hasCaption directly, once with the new typed-caption (releasing soon), the same problem of loosing the first caption would appear again if one chooses not to select any speechLocale in the audio-modal...

So, what I propose is we refactor the `captionSetOwner` (action responsible to insert something in the `caption_locale` table) that is not being used anymore, and whenever a person is going to start a caption (whatever the source is), we change the flow a bit to automatically send that action, this would mean that the caption_locale would be populated beforehand and the user could select what caption they would want to see. So to do this, for the core of BBB, we could simply send the speechLocale in this action when a person selects a language to speak, and for the plugins, we could warn the developers in the `serverCommand`(responsible for sending out a caption) ts-doc and in the readme file to send the locale that the caption is going to be previously.

Any thoughts on this? Feel free to comment or adjust anything you might find flawed

Reply:
I think we should abandon the concept of an "owner" for a caption locale. It was useful when typed captions relied on Pads, where only one user could type at a time.

However, this no longer makes sense.

I also don't like depending on the client sending a mutation to inform it that some user will send captions soon. It feels like over-engineering.

So, my suggestion is to observe `speechLocale` to predict that captions will be available. 
Additionally, introduce a `captionAddLocale` mutation (replacing `captionSetOwner`) to inform that someone will send captions for that language. This can be used by plugins. This approach is similar to your idea.

We would then observe both options to display the CC button.

What do you think?

Reply:
Sounds good to me! And yeah, when I said refactoring the owner, that was the main reason, it doesn't make much sense right now to have the "owner" of a caption, so the name `captionAddLocale` is a good one, thanks! I will implement those suggestions!